src/ledger-mobile-bridge.ts

```diff
@@ -0,0 +1,237 @@
+        'Ledger app is not initialized. You must call setTransport first.',
+      );
+    }
+    return Promise.resolve(this.app);
+  }
+
+  async initEthApp(): Promise<void> {
+    this.app = new LedgerHwAppEth(await this.getTransport());
+  }
+
+  async openEthApp(): Promise<void> {
+    const transport = await this.getTransport();
+    await transport.send(
+      0xe0,
+      0xd8,
+      0x00,
+      0x00,
+      Buffer.from(this.ethAppName, this.transportEncoding),
+    );
+  }
+
+  async closeApps(): Promise<void> {
+    const transport = await this.getTransport();
+    await transport.send(0xb0, 0xa7, 0x00, 0x00);
+  }
+
+  async getEthAppNameAndVersion(): Promise<GetEthAppNameAndVersionResponse> {
+    const transport = await this.getTransport();
+
+    const response = await transport.send(0xb0, 0x01, 0x00, 0x00);
+
+    let i = 1;
+    const format = response[i];
+
+    if (format !== 1) {
+      throw new Error('getEthAppNameAndVersion: format not supported');
+    }
+
+    i += 1;
+    const nameLength = response[i] ?? 0;
+    const appName = response
+      .slice(i, (i += nameLength))
+      .toString(this.transportEncoding);
+
+    i += 1;
+    const versionLength = response[i] ?? 0;
+    const version = response
+      .slice(i, (i += versionLength))
+      .toString(this.transportEncoding);
+
+    return {
+      appName,
+      version,
+    };
+  }
+}
+
+/**
+ * LedgerMobileBridge is a bridge between the LedgerKeyring and the LedgerMobileTransportMiddleware.
+ */
+export default class LedgerMobileBridge implements ILedgerMobileBridge {
+  transportMiddleware: ILedgerMobileTransportMiddleware;
+
+  deviceId = '';
+
+  isDeviceConnected = false;
+
+  constructor(opts?: LedgerMobileBridgeOptions) {
+    if (opts) {
+      this.setOptions(opts);
+    }
+    this.transportMiddleware = new LedgerMobileTransportMiddleware();
+  }
+
+  // init will be called by the eth keyring controller when new account added
+  async init(): Promise<void> {
+    return Promise.resolve();
+  }
+
+  setDeviceId(deviceId: string): void {
+    if (this.deviceId && this.deviceId !== deviceId) {
+      throw new Error('LedgerKeyring: deviceId mismatch.');
+    }
+    this.deviceId = deviceId;
+    this.isDeviceConnected = true;
+  }
+
+  getDeviceId(): string {
+    return this.deviceId;
+  }
+
+  async updateTransportMethod(): Promise<boolean> {
+    throw new Error('Method not supported.');
+  }
+
+  async attemptMakeApp(): Promise<boolean> {
+    throw new Error('Method not supported.');
+  }
+
+  async destroy(): Promise<void> {
+    await this.transportMiddleware.dispose();
+    this.isDeviceConnected = false;
+  }
+
+  async forgetDevice(): Promise<void> {
+    this.deviceId = '';
+    this.isDeviceConnected = false;
+  }
+
+  getOptions(): LedgerMobileBridgeOptions {
+    return {
+      deviceId: this.deviceId,
+    };
+  }
+
+  setOptions(opts: LedgerMobileBridgeOptions): void {
+    this.deviceId = opts.deviceId ?? '';
+  }
```

Comment:
what happens to the bridge is the `deviceId` here is changed to another value? do we need to take some side-actions or we are good to go?